aip/general/0127.md

```diff
@@ -92,6 +92,17 @@ message CreateBookRequest {
   - The `body` **must not** be a `repeated` field.
   - Fields **should not** use the `json_name` annotation to alter the field
     name in JSON, unless doing so for backwards-compatibility reasons.
+  - In any given HTTP request, if a field is represented in both the body
+    and the URI, the values **must** be identical. The server behavior when
+    this constraint is violated is unspecified.
+- Eligible fields which are not represented in the URI path or the body are mapped
+  to URI query parameters.
+  - Only singular primitive, repeated primitive, and singular message type fields
+    are eligible to be represented as query parameters.
```

Comment:
nit: Consider simpler wording: s/are eligible to be/may be/

Reply:
I'd prefer to keep it as "eligible" as "may" suggests something that's optional - a client might or might not decide to encode some fields as URI query parameters. In reality, a client *must* do that, because the fields are eligible to be encoded. Happy to noodle on alternatives to "eligible" - I'd just prefer to avoid "may".

Reply:
Oh, I see what you mean. On a naive read, "eligible" registered to me as something you can choose to do, i.e. "may". But you are using it in an algorithmic sense, tying back to your previous line.

Could we say instead:
```
Fields which are not represented in the URI path or the body are mapped to URI query parameters if and only if they 
are singular primitive, repeated primitive, or singular message type fields.
- Repeated primitive...
- For message-type fields....
```

And that also brings up a question to clarify: what about repeated message fields? Should it be an error for there to be an http binding that does not map them to the body, since they can't be represented in the path or query params and so would incur data loss if non-empty?